### PR TITLE
Fix #4007 : Backstap loop error in profile edit fragment error

### LINKED ISSUE #4007: ProfileEditFragment Backstack loop error
**Describe the bug**
In tablet When we reset or rename the Profile by Administrator controls in ProfileEditFragment and then on Pressing the back button we get stuck in a loop
**To Reproduce**
1. Go to Administrator Controls and click on any profile and then reset the pin or rename the name 
2. Click on the BackButton
3. Again Click on the back button
4. You get stuck in a loop of ProfileEditFragment and ProfileListFragment

**Expected behavior**
You should go back to administrator controls activity after pressing back button

**Demonstration**
|   |    |   |
| - | - | - |
| Screenshot_20211122_210529 | Screenshot_20211122_210618 | Screenshot_20211122_210644 |
| Screenshot_20211122_210701 | Screenshot_20211122_210706 |

https://user-images.githubusercontent.com/76404817/142893869-2f33edc5-0c89-4267-a5ff-af8bfe17f3e4.mp4

**Environment**
- Device/emulator being used: <device version/emulator>
- Android or SDK version : Android 9 Oreo/ api 28
- App version : latest dev

### PR DESCRIPTION
**Pull request by chrk2205**
## Explanation

Fixes #4007 
Instead of creating a new activity before submitting I am just finishing the present activity and going to the previous activity.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by yash10019coder** [comment]
Hi @chrk2205  please add the Fixes #<issue number> so that PR and issue gets linked

**Comment by chrk2205** [comment]
> Hi @chrk2205 please add the Fixes # so that PR and issue gets linked

i have included issue number.

**Review by yash10019coder — CHANGES_REQUESTED** [review]
Hi @chrk2205 overall idea is nice and this also works but fix the failing unit tests and also lint checks.
Thanks

**Comment by chrk2205** [comment]
@yash10019coder 
lint checks are passed can you help with the failing tests?
and also I am unable to run JUNIT tests on my android studio it throws some errors.
Thanks.

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt [hunk 1]
```diff
@@ -104,9 +103,7 @@ class ProfileRenameActivityPresenter @Inject constructor(
 
   private fun handleAddProfileResult(result: AsyncResult<Any?>, profileId: Int) {
     if (result.isSuccess()) {
-      val intent = ProfileEditActivity.createProfileEditActivity(activity, profileId)
```
Comment (yash10019coder) [inline app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt:107]: The hint is that we are no longer creating new Intent and launching the activity, instead we are using the previous intent.
So our test cases are written for checking for creating of new Intent, therefore learn how to test intents and then look in the tests of these activities.

**Review by yash10019coder — CHANGES_REQUESTED** [review]
@chrk2205 nice for fixing the lint checks you can see the failing roboelectirc tests and I have made a comment PTAL thanks

**Comment by yash10019coder** [comment]
and also if you are requesting review from someone please assign the person in the PR it makes it much easier.

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt (hunk 1 above)
Comment (chrk2205) [reply app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt:107]: `System.logW: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.java.lang.Throwable: Explicit termination method 'dispose' not called`
I tried to run the tests and the following error is thrown. The tests have no error but something else does. 
can you have a look at this?
@yash10019coder

Comment (yash10019coder) [reply app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt:107]: please add Screenshots of the error and the full error text

**Comment by yash10019coder** [comment]
Hi @chrk2205  this is just a warning and not a error and also please merge with the latest develop and also resolve the merge confilcts 
Thanks

**Comment by chrk2205** [comment]
@yash10019coder 
I am closing this PR I will update the project on my fork and keep a new PR. 
Thanks

**Comment by yash10019coder** [comment]
ok

**Pull request closed without merging**